test/parallel/test-timers-immediate-as-number.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const { test } = require('node:test');
+
+test('setImmediate return value is coerced to a number', (t) => {
+  const immediateId = +setImmediate(common.mustNotCall());
+  clearImmediate(immediateId);
+
+  t.assert.strictEqual(typeof immediateId, 'number');
+  t.assert.ok(!Number.isNaN(immediateId), 'setImmediate return value is NaN');
+});
+
+test('setImmediate return value is coerced to a string', (t) => {
+  const immediateId = '' + setImmediate(common.mustNotCall());
+  clearImmediate(immediateId);
+
+  t.assert.strictEqual(typeof immediateId, 'string');
+  t.assert.ok(!Number.isNaN(immediateId), 'setImmediate return value is NaN');
```

Comment:
`!Number.isNan(string)` is always `true`, we have to test it differently. For example:
```suggestion
  t.assert.strictEqual(typeof immediateId, 'string');
  t.assert.match(immediateId, /^\d+$/, 'setImmediate return value is not non-negative integer');
```